martin/src/sprites/mod.rs

```diff
@@ -186,14 +187,19 @@ async fn parse_sprite(
     let tree = Tree::from_data(&buffer, &Options::default())
         .map_err(|e| SpriteParsingError(e, path.clone()))?;
 
-    let sprite = Sprite::new(tree, pixel_ratio).ok_or_else(|| SpriteInstError(path.clone()))?;
+    let sprite = if as_sdf {
+        Sprite::new_sdf(tree, pixel_ratio).ok_or_else(|| SpriteInstError(path.clone()))?
+    } else {
+        Sprite::new(tree, pixel_ratio).ok_or_else(|| SpriteInstError(path.clone()))?
+    };
```

Comment:
The reason why I left this duplicated is that I don't think that the alternatives below are better ^^

here are the alternatives:

```suggestion
    let new_sprite = if as_sdf { Sprite::new_sdf } else { Sprite::new };
    let sprite = new_sprite(tree, pixel_ratio).ok_or_else(|| SpriteInstError(path.clone()))?;
```
```suggestion
    let sprite = if as_sdf {
        Sprite::new_sdf(tree, pixel_ratio)
    } else {
        Sprite::new(tree, pixel_ratio)
    }
    .ok_or_else(|| SpriteInstError(path.clone()))?;
```

```suggestion
    let sprite = if as_sdf {
        Sprite::new_sdf(tree, pixel_ratio)
    } else {
        Sprite::new(tree, pixel_ratio)
    };
    let sprite = sprite.ok_or_else(|| SpriteInstError(path.clone()))?;
```

Reply:
thx, slightly shorter now.  I wasn't even thinking of the first option, nice, but too magical :)  I generally prefer consolidated error handling when possible, but not too crazy (whatever that even means )